src/Main/Settings.h

```diff
@@ -0,0 +1,44 @@
+// sma find where to get current copyright text from
+
+#ifndef SETTINGS_H
+#define SETTINGS_H
+
+#include <QJsonObject>
+#include <QJsonValue>
+
+namespace FOEDAG {
+
+class Settings {
+ private:
+  Settings();
+
+  static Settings* m_instance;
+  QJsonObject m_data;
+
+ public:
+  static Settings* getInstance();
+  static void setValue(QJsonObject& parent, const QString& key,
```

Comment:
what's the point to have all methods static?

Reply:
I'm not a fan of singletones, there is always problems when you need write some unit tests. And in general it's bad practice.

Reply:
Oh interesting, yeah i'm open to whatever is best. Let's discuss in the the sync up.

Reply:
Gave our existing code a look and gave this some more thought. What if I change Settings to a normal class (remove singleton and static elements), and then instantiate during bootstrap and store the object in the Session object similar to how TclInterpreter, CommandLine, CommandStack, etc are doing it? This should address the single instance issues you foresaw while keeping the settings w/ 1 primary copy of the data which I think is important to maintain as more systems begin to read/write to the settings.